packages/@tinacms/schema-tools/src/schema/addNamespaceToSchema.test.ts

```diff
@@ -0,0 +1,352 @@
+      children: [
+        { name: 'child1', namespace: ['child1'] },
+        {
+          name: 'child2',
+          namespace: ['child2'],
+          children: [
+            { name: 'child1', namespace: ['child2', 'child1'] }, // Should not be affected by the first 'child1'
+          ],
+        },
+      ],
+    })
+
+    // Ensure namespaces do not share references
+    expect((result.children[0] as Node).namespace).toEqual(['child1'])
+    expect((result.children[1] as Node).namespace).toEqual(['child2'])
+    expect((result.children?.[1]?.children?.[0] as Node).namespace).toEqual([
+      'child2',
+      'child1',
+    ])
+  })
+
+  test('should generate unique namespaces for nested nodes with similar names', () => {
+    const node = {
+      name: 'root',
+      children: [
+        { name: 'child', children: [{ name: 'grandchild' }] },
+        { name: 'child', children: [{ name: 'grandchild' }] }, // Same name structure
+      ],
+    }
+
+    const result = addNamespaceToSchema(node)
+
+    // Verify that namespaces are correctly separated and unique
+    expect(result).toEqual({
+      name: 'root',
+      namespace: [],
+      children: [
+        {
+          name: 'child',
+          namespace: ['child'],
+          children: [
+            { name: 'grandchild', namespace: ['child', 'grandchild'] },
+          ],
+        },
+        {
+          name: 'child',
+          namespace: ['child'],
+          children: [
+            { name: 'grandchild', namespace: ['child', 'grandchild'] },
+          ],
+        },
+      ],
+    })
+
+    // Ensure namespaces do not share references and are independently constructed
+    expect((result.children[0] as Node).namespace).not.toBe(
+      (result.children[1] as Node).namespace
+    )
+    expect((result.children[0].children[0] as Node).namespace).not.toBe(
+      (result.children[1].children[0] as Node).namespace
+    )
+  })
+
+  test('should correctly handle repeated names at different levels', () => {
+    const node = {
+      name: 'root',
+      children: [
+        {
+          name: 'child',
+          nested: { name: 'child' }, // Repeated name at a deeper level
+        },
+      ],
+    }
+
+    const result = addNamespaceToSchema(node)
+
+    // Verify that namespaces are correctly constructed
+    expect(result).toEqual({
+      name: 'root',
+      namespace: [],
+      children: [
+        {
+          name: 'child',
+          namespace: ['child'],
+          nested: {
+            name: 'child',
+            namespace: ['child', 'child'], // Should reflect the nesting with repeated name
+          },
+        },
+      ],
+    })
+
+    // Ensure nested namespaces are correctly formed
+    expect((result.children[0] as Node).namespace).toEqual(['child'])
+    expect((result.children[0].nested as Node).namespace).toEqual([
+      'child',
+      'child',
+    ])
+  })
+
+  test('should preserve existing namespaces for repeated names at different levels', () => {
```

Comment:
Are "should preserve existing namespaces for repeated names at different levels" and "should preserve existing namespaces and not modify them" not testing for the same thing? That existing namespaces should not be modified? Could these be reduced into a single test?

Reply:
👍🏻 removed duplicate test